src/coreComponents/mesh/CellElementSubRegion.hpp

```diff
@@ -247,13 +329,76 @@ class CellElementSubRegion : public CellBlock
     } );
   }
 
+  void calculateElementGeometricQuantities( NodeManager const & nodeManager,
+                                            FaceManager const & faceManager ) override;
+
+private:
+
   /// Map used for constitutive grouping
   map< string, localIndex_array > m_constitutiveGrouping;
 
   /// Array of constitutive point volume fraction
   array3d< real64 > m_constitutivePointVolumeFraction;
 
-private:
+  /// Element-to-node relation
+  NodeMapType m_toNodesRelation;
+
+  /// Element-to-edge relation
+  EdgeMapType m_toEdgesRelation;
+
+  /// Element-to-face relation
+  FaceMapType m_toFacesRelation;
+
+  /// Name of the properties registered from an external mesh
+  string_array m_externalPropertyNames;
+
+  /**
+   * @brief Compute the volume of the k-th element in the subregion.
+   * @param[in] k the index of the element in the subregion
+   * @param[in] X an arrayView of (const) node positions
+   */
+  inline void calculateCellVolumesKernel( localIndex const k,
+                                          arrayView2d< real64 const, nodes::REFERENCE_POSITION_USD > const & X ) const
+  {
+    LvArray::tensorOps::fill< 3 >( m_elementCenter[ k ], 0 );
+
+    real64 Xlocal[10][3];
```

Comment:
Does that contain an assumption on the max number of nodes per element? Do you think we could remove the hardcoded value of 10 and replace it with a `constexpr` defined at the top of the class to make the transition to polyhedral grids easier?